azurerm/resource_arm_action_group.go

```diff
@@ -0,0 +1,313 @@
+		parameters.ActionGroup.EmailReceivers = expandActionGroupEmailReceiver(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("sms_receiver"); ok {
+		parameters.ActionGroup.SmsReceivers = expandActionGroupSmsReceiver(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("webhook_receiver"); ok {
+		parameters.ActionGroup.WebhookReceivers = expandActionGroupWebHookReceiver(v.([]interface{}))
+	}
+
+	_, err := client.CreateOrUpdate(ctx, resGroup, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error creating or updating action group %s (resource group %s): %+v", name, resGroup, err)
+	}
+
+	read, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error getting action group %s (resource group %s) after creation: %+v", name, resGroup, err)
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Action group %s (resource group %s) ID is empty", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmActionGroupRead(d, meta)
+}
+
+func resourceArmActionGroupRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).actionGroupsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error parsing action group resource ID \"%s\" during get: %+v", d.Id(), err)
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["actionGroups"]
+
+	resp, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		if response.WasNotFound(resp.Response.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error getting action group %s (resource group %s): %+v", name, resGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	d.Set("short_name", *resp.GroupShortName)
+	d.Set("enabled", *resp.Enabled)
+
+	if err = d.Set("email_receiver", flattenActionGroupEmailReceiver(resp.EmailReceivers)); err != nil {
+		return fmt.Errorf("Error setting `email_receiver` of action group %s (resource group %s): %+v", name, resGroup, err)
+	}
+
+	if err = d.Set("sms_receiver", flattenActionGroupSmsReceiver(resp.SmsReceivers)); err != nil {
+		return fmt.Errorf("Error setting `sms_receiver` of action group %s (resource group %s): %+v", name, resGroup, err)
+	}
+
+	if err = d.Set("webhook_receiver", flattenActionGroupWebHookReceiver(resp.WebhookReceivers)); err != nil {
+		return fmt.Errorf("Error setting `webhook_receiver` of action group %s (resource group %s): %+v", name, resGroup, err)
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmActionGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).actionGroupsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error parsing action group resource ID \"%s\" during delete: %+v", d.Id(), err)
```

Comment:
we don't need to wrap this error message - so can we just return it directly?

Reply:
we consistently across the codebase don't wrap these error messages since they're an internal error users aren't going to be able to work around (vs a temporary issue in the Azure API in CRUD/Polling where they may be able to self-diagnose) - as such ID parsing error don't need wrapping.